.github/workflows/scripts/comments.mjs

```diff
@@ -1,14 +1,96 @@
+import { readFile } from 'node:fs/promises'
+
+/**
+ * Posts the content of multiple diffs in parallel on the given Github issue
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {DiffComment[]} diffs
+ */
+export async function commentDiffs (
+  githubActionContext,
+  issueNumber,
+  diffs
+) {
+  // Run the comments in parallel to avoid that an early one failing prevents the others
+  // and use `allSettled` to avoid the promise rejecting on the first failure but wait
+  // for everything to complete
+  const results = await Promise.allSettled(
+    diffs.map((diff) =>
+      commentDiff(githubActionContext, issueNumber, diff)
+    )
+  )
+
+  const resultsWithErrors = results.filter(result => result.status === 'rejected')
+  if (resultsWithErrors.length) {
+    throw new Error('Posting diff comment failed', { cause: resultsWithErrors })
```

Comment:
This is nice too, love using error causes

Is there friendlier error info by grabbing the `{ reason }` property off all those rejected promises?

```js
if (results.some(({ status }) => status === 'rejected')) {
 // … Map each `{ reason }` property from `PromiseRejectedResult`
}

Reply:
It'll likely be less verbose as we don't really need to know the `status` is rejected in the logs. I'll do a quick mapping there 😊

Reply:
Actually no, I think we might need the verbosity there and my `filter` is getting in the way of us understanding what broke. The `filter` makes us lose the index in the array of rejected Promises, so we don't know which comment failed. We'll see from Github UI, but we may as well dump the whole `results` in the logs to avoid back and forth between the two views 😊

Reply:
Yeah, now we know which posting failed from the log:

```
Error: Posting diff comment failed
    at commentDiffs (file:///home/runner/work/govuk-frontend/govuk-frontend/.github/workflows/scripts/comments.mjs:25:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: Unhandled error: Error: Posting diff comment failed
    at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16), <anonymous>:27:1)
    at async main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15236:20) {
  [cause]: [
    { status: 'fulfilled', value: undefined },
    {
      status: 'rejected',
      reason: Error
          at commentDiff (file:///home/runner/work/govuk-frontend/govuk-frontend/.github/workflows/scripts/comments.mjs:62:47)
    },
    { status: 'fulfilled', value: undefined }
  ]
}
```

That also solves any typing issue I believe as we're not manipulating the results, just dumping them 😊